Importer/jctdata/jac.py

```diff
@@ -133,9 +133,40 @@ def issn_clusters(coincident_issn_files, clusters_file):
     issn_clusters.sort()
     issn_clusters = list(k for k,_ in itertools.groupby(issn_clusters))
 
+    d = {}
+    for row in issn_clusters:
+        if row[0] not in d:
+            d[row[0]] = []
+        d[row[0]] += [row[x + 1] for x in range(len(row) - 1)]
+
+    # go through every key, and for each value in the array associated with it, look
+    # for the key of that value.  This will lead to another set of values.  For each
+    # of those transitive values that are not the original key and are not in the
+    # original list of values, add it to a list of additional values to add.  Then remove
+    # the key of the value, so we don't count it again.
+    #
+    # In this way, we "consume" all the values into their unique super-clusters by
+    # expanding the first list of values with values from all transitive lists.
+    # In theory we end up with a unique list of coincident identifiers, with all
+    # smaller subsets of the larger cluster subsumed into one.
+    keys = list(d.keys())
```

Comment:
high memory use - if we're not using the value of `keys` we should just use the iterator without casting to list

Reply:
We have to cast to a list here because otherwise we get a concurrent modification error in the code block below

Reply:
Specifically line 160, which deletes an element from the dictionary.  Because the iterator is bound to the dictionary, that triggers the concurrent modification error.  Casting to a list dereferences that key iterator

Reply:
ah, right! thanks